src/components/borrow/borrow.service.js

```diff
@@ -0,0 +1,135 @@
+    }
+
+    if (bookId.length !== 24) {
+      const error = new Error("INVALID_BOOK_ID");
+      throw error;
+    }
+    const bookFound = await bookMdl.book.findById(bookId);
+    if (!bookFound) {
+      const error = new Error("BOOK_NOT_FOUND");
+      throw error;
+    }
+    if (bookFound.availableCopies === 0) {
+      const error = new Error("BOOK_NOT_AVAILABLE");
+      throw error;
+    }
+
+    const borrowBook = await borrowMdl.borrow({
+      bookId: bookId,
+      userId: userId,
+      borrowDate: new Date().toISOString(),
+      dueDate: helper.calculateDueDate(new Date()),
+      returnDate: null,
+      status: "borrowed",
+      fine: 0,
+    });
+    const bookBorrowed = await borrowBook.save();
+
+    bookFound.availableCopies -= 1;
+    await bookFound.save();
+
+    return bookBorrowed;
+  } catch (err) {
+    const error = new Error(err.message);
+    throw error;
+  }
+};
+
+const returnBook = async (req) => {
+  const { borrowId } = req.body;
+  try {
+    const borrowRecordFound = await borrowMdl.borrow
+      .findById(borrowId)
+      .populate("bookId");
+
+    if (!borrowRecordFound) {
+      const error = new Error("NO_HISTORY");
+      throw error;
+    }
+    const bookData = borrowRecordFound.bookId;
+    if (bookData.availableCopies < bookData.totalCopies) {
+      bookData.availableCopies += 1;
+      await bookData.save();
+    } else {
+      const error = new Error("BOOK_NOT_BORROWED");
+      throw error;
+    }
+    if (borrowRecordFound.userId.toString() !== req.user._id) {
+      const error = new Error("YOU_NOT_BORROWED");
+      throw error;
+    }
+    if (
+      borrowRecordFound.returnDate &&
+      borrowRecordFound.status === "returned"
+    ) {
+      const error = new Error("BOOK_RETURNED");
+      throw error;
+    }
+
+    const dueDate = borrowRecordFound.dueDate;
+    const returnDate = new Date();
+    const fine = helper.calculateFine(dueDate, returnDate);
+    const bookReturned = await borrowMdl.borrow.findByIdAndUpdate(
+      borrowId,
+      {
+        returnDate: new Date().toISOString(),
+        status: "returned",
+        fine: fine,
+      },
+      { new: true }
+    );
+
+    return bookReturned;
+  } catch (err) {
+    const error = new Error(err.message);
+    throw error;
+  }
+};
+
+const extendBorrowing = async (req) => {
+  const { borrowId } = req.body;
+  try {
+    if (borrowId.length !== 24) {
+      const error = new Error("INVALID_BORROW_ID");
+      throw error;
+    }
+    const borrowedData = await borrowMdl.borrow.findOne({ _id: borrowId });
+    if (borrowedData.userId.toString() !== req.user._id) {
+      const error = new Error("This book is not borrowed by you");
+      throw error;
+    }
+    if (!borrowedData) {
+      const error = new Error("NO_HISTORY");
+      throw error;
+    }
+    const extendTime = await borrowMdl.borrow.findByIdAndUpdate(
+      borrowId,
+      {
+        dueDate: helper.extendDueDate(borrowedData.dueDate),
```

Comment:
jyare pan database query kare tyare ama dynamic function call nai karvano.
pehla upar value lai aav, pachi ae value ama nakh.

so, `const someVariable = helper.extendDueDate(borrowedData.dueDate)`
then 
`dueDate: someVariable`

Reply:
alright

Reply:
Ama error ave 6 avi rite pela variable ma function call kravine data save rkhu to kyrek time ma difference ave 6